terms.go

```diff
@@ -82,19 +91,14 @@ func (e *Env) Model() G.Nodes {
 	return retVal
 }
 
+// HintedModel will return the gorgonia.Nodes hinted associated with this environment
 func (e *Env) HintedModel(hint int) G.Nodes {
 	prealloc := make(G.Nodes, 0, hint)
 	e.hinted(prealloc)
 	return prealloc
 }
 
-func (e *Env) hinted(prealloc G.Nodes) {
```

Comment:
Why is this removed? `HintedModel` calls `hinted`

Reply:
It was moved to the top, that way private methods and public methods are grouped together :)